src/ping.ts

```diff
@@ -38,21 +41,21 @@ export class Ping {
 
   private async performPing (): Promise<void> {
     try {
-      await this.client._stanzaio.ping(this.options.jid);
+      await this.stanzaInstance.ping(this.options.jid);
       this.numberOfFailedPings = 0;
       this.queueNextPing();
     } catch (err) {
       const info = {
         channelId: this.client.config.channelId,
-        jid: this.client._stanzaio.jid
+        jid: this.stanzaInstance.jid
       };
       this.client.logger.warn('Missed a ping.', Object.assign({ error: err }, info));
 
       /* if we have reached max number of missed pings, disconnect */
       if (++this.numberOfFailedPings > this.failedPingsBeforeDisconnect) {
-        this.stop();
         this.client.logger.error('Missed too many pings, disconnecting', Object.assign({ numberOfFailedPings: this.numberOfFailedPings }, info));
-        this.client._stanzaio.sendStreamError({ text: 'too many missed pongs', condition: 'connection-timeout' });
+        this.stanzaInstance.sendStreamError({ text: 'too many missed pongs', condition: 'connection-timeout' });
```

Comment:
Is "pong" an actual term???

Reply:
lol. frankly i have no idea. makes sense to me though.

Reply:
At least for WebSockets, [it is](https://www.rfc-editor.org/rfc/rfc6455#section-5.5.3). 😄